cmd/zfs/zfs_main.c

```diff
@@ -4145,16 +4149,17 @@ zfs_do_send(int argc, char **argv)
 			fromname = frombuf;
 		}
 		err = zfs_send_one(zhp, fromname, STDOUT_FILENO, flags);
+		free(fromname);
```

Comment:
This could free `frombuf` which is on the stack.  This might be what caused the ZTS failure.